src/radical/pilot/agent/executing/base.py

```diff
@@ -50,16 +50,16 @@ def create(cls, cfg, session):
         if cls != AgentExecutingComponent:
             raise TypeError('Factory only available to base class!')
 
-        name = cfg['spawner']
+        name = session._reg['rcfg.agent_spawner']
```

Comment:
we should take this from `..._reg['cfg.spawner']`, since config might not set this, and PMGRLaunching will set a default value

https://github.com/radical-cybertools/radical.pilot/blob/a4caa5e6b23b94b99baa329b02bf33dacdf9dab8/src/radical/pilot/pmgr/launching/base.py#L565

https://github.com/radical-cybertools/radical.pilot/blob/a4caa5e6b23b94b99baa329b02bf33dacdf9dab8/src/radical/pilot/pmgr/launching/base.py#L859

Reply:
I know we discussed this - but for now I think I would like to leave as is after all.  Let's clean this up once we have a config schema...